src/trg-main-window.c

```diff
@@ -2483,42 +2402,18 @@ static void trg_main_window_set_hidden_to_tray(TrgMainWindow * win,
     priv->hidden = hidden;
 }
 
-static gboolean
-window_state_event(TrgMainWindow * win,
-                   GdkEventWindowState * event, gpointer trayIcon)
-{
-    TrgMainWindowPrivate *priv = trg_main_window_get_instance_private(win);
-    TrgPrefs *prefs = trg_client_get_prefs(priv->client);
-
-    if (priv->statusIcon
-        && (event->changed_mask & GDK_WINDOW_STATE_ICONIFIED)
-        && (event->new_window_state & GDK_WINDOW_STATE_ICONIFIED)
-        && trg_prefs_get_bool(prefs, TRG_PREFS_KEY_SYSTEM_TRAY_MINIMISE,
-                              TRG_PREFS_GLOBAL)) {
-        trg_main_window_set_hidden_to_tray(win, TRUE);
-        return TRUE;
-    }
-
-    return FALSE;
-}
 
-void trg_main_window_remove_status_icon(TrgMainWindow * win)
+void trg_main_window_remove_tray(TrgMainWindow * win)
 {
-    TrgMainWindowPrivate *priv = trg_main_window_get_instance_private(win);
 #if HAVE_LIBAPPINDICATOR
-    if (priv->appIndicator) {
-        g_object_unref(G_OBJECT(priv->appIndicator));
+    TrgMainWindowPrivate *priv = trg_main_window_get_instance_private(win);
 
-        priv->appIndicator = NULL;
-    } else {
-#else
-    if (1) {
-#endif
-        if (priv->statusIcon)
-            g_object_unref(G_OBJECT(priv->statusIcon));
+    if (priv->appIndicator)
+        app_indicator_set_status(priv->appIndicator,
+                                 APP_INDICATOR_STATUS_PASSIVE);
```

Comment:
Explicitly setting this before removing the object didn't work either. I'm out of ideas....

Reply:
Could it be the remaining reference to the menu?

Reply:
Nope, that didn't work either.